be/src/olap/page_cache.h

```diff
@@ -28,11 +28,55 @@
 
 #include "olap/lru_cache.h"
 #include "util/slice.h"
+#include "vec/common/allocator.h"
+#include "vec/common/allocator_fwd.h"
 
 namespace doris {
 
 class PageCacheHandle;
 
+template <typename TAllocator>
+class PageBase : private TAllocator {
+public:
+    PageBase() : data(nullptr), size(0), bytes(0) {}
+
+    PageBase(size_t b) : size(b), bytes(b) {
+        data = reinterpret_cast<char*>(TAllocator::alloc(bytes, ALLOCATOR_ALIGNMENT_16));
+        ExecEnv::GetInstance()->page_no_cache_mem_tracker()->consume(bytes);
+    }
+
+    PageBase(PageBase&& other) : data(other.data), size(other.size) {
+        std::swap(bytes, other.bytes);
+    }
+
+    ~PageBase() {
+        if (data != nullptr) {
+            DCHECK(bytes != 0 && size != 0);
+            TAllocator::free(data, bytes);
+            ExecEnv::GetInstance()->page_no_cache_mem_tracker()->release(bytes);
+        }
+    }
+
+    void reset_size(size_t n) {
+        DCHECK(n <= bytes);
+        size = n;
+    }
+
+    void release() {
```

Comment:
is this method useful?

Reply:
fix

Reply:
for page_cache_test.cpp, but it can be optimized